### DISCUSSION TITLE
Support for the Commodore PC 10/20-III line (maybe Colt/PC-1 too)

### DISCUSSION THREAD

**Discussion by shakeyourbunny** [86Box/86Box#3360] [Ideas]
**Is your feature request related to a problem? Please describe.**
The Commodore PC-10-III computer model was removed by 86box (PCem does support a specific BIOS version)

It would be really neat to support the Commodore PC 10-III and PC 20-III.

Specifications:
- CPU: Intel 8088-1 running at 4.77MHz (Standard), 7.16Mhz (Turbo), or 9.54MHz (Double)
- Optional 8087 math co-processor
- RAM: 640KB
- ROM: 32KB Phoenix Self-configuring BIOS (Versions 4.35 through 4.41 are known to exist) http://ftp.zimmers.net/anonftp/pub/cbm-pc/firmware/pc10/index.html
- Chipset: Faraday FE2010A single chip controller IC (see https://github.com/skiselev/micro_8088/blob/master/Documentation/Faraday-XT_Controller-FE2010A.md, https://dosdays.co.uk/topics/faraday_chipset.php)
- Clock: Battery-backed Real Time Clock (RTC) maintains the date and time while the system is powered off
- Floppy Drive: One or Two 360K 5.25-inch (720K 3.5-inch optional on PC20-III) (SMC FDC9268 controller)
- Hard Drive: 20MB XTA (8-bit IDE) (standard on PC20-III)
- Sound: On-board piezoelectric beeper

- Ports:
9-pin d-sub Bus Mouse port compatible with Commodore Amiga 1352 mouse (MOS 5720)
25-pin d-sub RS-232 serial port (Western Digital WD8250-PL)
25-pin d-sub 8-bit Centronics parallel port (Paradise PPC1)
Din-5 IBM PC XT-compatible keyboard port
9-pin d-sub RGBI/Monochrome video port
RCA composite video port

- Video: Onboard Paradise PVC4 ( see https://bitsavers.org/components/westernDigital/_dataSheets/PVC4.pdf )
MDA Monochrome Text
Hercules Monochrome Graphics
CGA Color Text and Graphics
Plantronics Colorplus Graphics
Alpha132 Monochrome Text

- 3 8-bit ISA slots for expansion
- 75 watt power supply

Full specs and description see http://ftp.zimmers.net/anonftp/pub/cbm-pc/documents/PC-8088-Information.txt 
Note that this computer can switch speed between 4,77/7,16/9,54 Mhz either with a utility or key combinations (ctrl+alt+s/t/d)
Full set of graphics capabilites is also documented there.

Technical manual for the PC 10-III/20-III: http://ftp.zimmers.net/anonftp/pub/cbm-pc/schematics/Commodore%20PC_10_III%20Technical%20Manual.pdf
Technical manual for the PC-1: http://ftp.zimmers.net/anonftp/pub/cbm-pc/schematics/Commodore%20PC-1%20Technical%20Manual.pdf
Technical manual for the Colt: http://ftp.zimmers.net/anonftp/pub/cbm-pc/schematics/Commodore%20PC-1%20Technical%20Manual.pdf
Note that for accurate emulation the graphics adapter DIP switch selection should be possible (for eg an external ISA video card instead of onboard graphics).

Runs with Commodore MSDOS 3.2x / 3.30 (see http://ftp.zimmers.net/anonftp/pub/cbm-pc/software/index.html )
It would be nice if it would support the available BIOS ROMs revisions ( http://ftp.zimmers.net/anonftp/pub/cbm-pc/firmware/pc10/index.html )

https://rclassiccomputers.com/2022/03/27/commodore-pc20-iii/

Maybe support for the Commodore COLT and/or the PC-1 (a cut down version) would be nice.

**Additional context**

Pictures and additional information:
https://rclassiccomputers.com/wp-content/uploads/2022/03/F93C1CEA-8702-4E1B-8691-1C1F8AA5CB15.jpeg
https://ancientelectronics.files.wordpress.com/2015/06/colt11.jpg

https://oldcomputer.info/pc/commodore20/

https://www.youtube.com/watch?v=3yyYmhvKmw0

**Comment by johansmolinski** [comment]
I very much second this request, but I would also love to see series I and II of the Commodore PC 10/20.

**Comment by Rdthe80sman** [comment]
When did 86box include the PC10-III?

**Reply by shakeyourbunny** [reply]
the PC10-III is basically the PC20-III, but with 2 360kb disk drives instead of the 20MB HDD + 1 disk drive.

**Comment by shakeyourbunny** [comment]
Any interest in implementing the Commodore PC10/20-III for 86box? 

It shouldn't be much work, as most things are standard components in these systems, but do have some unique features.

**Reply by jriwanek** [reply]
We're certainly *interested*, but we're also currently in a machine freeze preparing for the next release.

**Reply by pankoza2-pl** [reply]
> We're certainly _interested_, but we're also currently in a machine freeze preparing for the next release.

You can create a separate branch like it was with version/4.2 which contained features withheld from the main branch due to the previous feature freeze

**Comment by Maverick-Shark** [comment]
Hi,
Any update? How did you solve booting from a big HDD (with ide-xt rom)? I reed Commodore PC motherboards have a jumper to enable it.

Thanks and best regards,
Shark

**Comment by jriwanek** [comment]
I can think of no reason Commodore PC motherboards would have a Jumper to enable support for a thing which was made decades after they were, could you be clearer?
